__tests__/app.test.js

```diff
@@ -199,6 +199,77 @@ describe('/api/articles/:article_id/comments', () => {
                 expect(response.body.msg).toBe('Article does not exist')
             })
     });
+    test('POST:200 adds comment and responds with posted comment', () => {
+        const newComment = {
+            'username': 'jessjelly',
+            'body': 'It were reet'
+        };
+        return request(app)
+            .post('/api/articles/3/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({body}) => {
+                expect(body.comment[0]).toEqual(
+                    expect.objectContaining({
+                        comment_id: expect.any(Number),
+                        body: 'It were reet',
+                        votes: expect.any(Number),
+                        author: 'jessjelly',
+                        article_id: 3,
+                        created_at: expect.any(String),
+                    })
+                );
+            });
+    });
+    test('POST:400 sends error message when given an invalid article id', () => {
+        const newComment = {
+            'username': 'jessjelly',
+            'body': 'It were reet'
+        };
+        return request(app)
+        .post('/api/articles/not-an-article/comments')
+        .send(newComment)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('Invalid ID');
+            });
+    });
+    test('POST:400 sends error message when given empty object', () => {
+        const blankUpdate = {};
+        return request(app)
+            .post('/api/articles/3/comments')
+            .send(blankUpdate)
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe("Comment undefined")
+            });
+    });
+    test('POST:404 sends error message when given a valid but non-existent id', () => {
+        const newComment = {
+            'username': 'jessjelly',
+            'body': 'It were reet'
+        };
+        return request(app)
+            .post('/api/articles/3141/comments')
+            .send(newComment)
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe('ID does not exist')
+            })
+    });
+    test('POST:404 sends error message when given non-existent username', () => {
+        const newComment = {
+            'username': 'nookapocalypse',
+            'body': 'It were reet'
+        };
+        return request(app)
+            .post('/api/articles/3/comments')
+            .send(newComment)
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe('ID does not exist')
+            })
+    });
```

Comment:
what if username or body were missing - I know you have the test on line 237 for an empty object, would that work for just one missing property too?